src/app/components/visualize/visualize.menus.ts

```diff
@@ -0,0 +1,36 @@
+import { Category, menuRoute } from "@interfaces/menusInterfaces";
+import { StrongRoute } from "@interfaces/strongRoute";
+
+export const visualizeRoute = StrongRoute.newRoot().add(
+  "visualize",
+  ({ siteId, regionId, projectId, extent0, extent1, lane }) => {
+    const qsp = {
+      extent0,
+      extent1,
+      lane,
+    };
+
+    if (siteId) {
+      return { siteId, ...qsp };
+    } else if (regionId) {
+      return { regionId, ...qsp };
+    } else if (projectId) {
+      return { projectId, ...qsp };
+    } else {
+      return qsp;
+    }
+  }
+);
+
```

Comment:
Missing QSP `siteIds : Array<Id>`

While flat parameters are the literal correct choice, this might feel better if implemented as a discriminated union for the parameters `siteId`, `siteIds`, `projectIds`, and `regionId` because they are exclusive choices.

Additionally, `regionId` is not currently supported by the old client... so perhaps throw error if we try to form such a link? Or better yet rewrite usages of `regionId` into `siteIds` QSP.